src/AppRoutes.jsx

```diff
@@ -0,0 +1,28 @@
+import React from "react";
+import { Routes, Route, Navigate, BrowserRouter } from "react-router-dom";
+import LandingPage from "./pages/LandingPage";
+import HomePage from "./pages/HomePage";
+import LoginPage from "./pages/LoginPage";
+import { useContext } from "react";
+import { UserContext } from "./contexts/UserContext";
+
+function AppRoutes() {
+  const { userInfo } = useContext(UserContext);
+  return (
+    <BrowserRouter>
+      <Routes>
+        <Route
+          path="/"
+          element={userInfo == null ? <LandingPage /> : <Navigate to="/homepage" />}
```

Comment:
If userInfo is not null, it might still be beneficial to check if it’s an empty object or another falsy value, depending on how userInfo is structured.